elftools/ehabi/decoder.py

```diff
@@ -0,0 +1,271 @@
+# -------------------------------------------------------------------------------
+# elftools: ehabi/decoder.py
+#
+# Decode ARM exception handler bytecode.
+#
+# LeadroyaL (leadroyal@qq.com)
+# This code is in the public domain
+# -------------------------------------------------------------------------------
+
+
+class EHABIBytecodeDecoder(object):
+    """ Decoder of a sequence of ARM exception handler abi bytecode.
+
+        Reference:
+        https://github.com/llvm/llvm-project/blob/master/llvm/tools/llvm-readobj/ARMEHABIPrinter.h
+        https://developer.arm.com/documentation/ihi0038/b/
+
+        Accessible attributes:
+
+            mnemonic_array:
+                before decode, None.
+                after decode, MnemonicItem array.
+
+        Parameters:
+
+            bytecode_array:
+                Integer array, raw data of bytecode.
+
+            auto_decode:
+                bool, whether bytecode should be decoded during construction.
+                (default : true)
+    """
+
+    def __init__(self, bytecode_array, auto_decode=True):
+        self._bytecode_array = bytecode_array
+        self._index = None
+        self.mnemonic_array = None
+        if auto_decode:
+            self.decode()
```

Comment:
I feel like `decode` should just be merged into `__init__`.  You're probably not going to redecode an `EHABIBytecodeDecoder` once constructed.

Reply:
I use `auto_decode` for developer who doesn't care the disasm result of bytecode. It's ok to merge, but I prefer to give the choice to developer.

Reply:
Can you elaborate on the scenario in which a developer would want to create an `EHABIBytecodeDecoder` without decoding the bytecode?

Reply:
I consider deeply and cannot imagine that scenario, I'll remove `auto_decode`. Thanks.